### PR TITLE
Grep

### PR DESCRIPTION
**Pull request by annaoliinyk**
https://exercism.org/tracks/python/exercises/grep

Search a file for lines matching a regular expression pattern. Return the line number and contents of each matching line.

Implement the Unix grep command

### CODE REVIEW

File: grep/grep_task.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                            non_matching_lines += files[i] + ":" + str(line_counter) + ":" + line
+                        else:
+                            non_matching_lines += files[i] + ":" + line
+                line_counter += 1
+
+    if "-l" not in flags and "-v" not in flags:
+        return result
+    elif "-v" in flags:
+        return non_matching_lines
+
```
Comment (AnaKovaGH) [inline grep/grep_task.py]: there is no need for this space

File: grep/grep_task.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    if len(files) == 1:
+        datafile = get_data_from_file(files[0])
+        line_counter = 1
+        for line in datafile:
+            result, non_matching_lines = get_matching_lines_single_file(line_counter, line, pattern, result, flags,
+                                                                        non_matching_lines)
+            line_counter += 1
+    elif len(files) > 1:
+        for i in range(len(files)):
+            datafile = get_data_from_file(files[i])
```
Comment (AnaKovaGH) [inline grep/grep_task.py]: Why do you open file with every loop run?

File: grep/grep_task.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            line_counter = 1
+            for line in datafile:
+                line_to_match = line
+                if "-i" in flags:
```
Comment (AnaKovaGH) [inline grep/grep_task.py]: Remove hardcoded values here. Make a separate constant outside with all this values.

**Review by AnaKovaGH — CHANGES_REQUESTED** [review]
Please also change that mane if-else statements

**Comment by annaoliinyk** [comment]
Added files needed for tests (from excercism)

**Comment by annaoliinyk** [comment]
Ready for review

**Review by AnaKovaGH — CHANGES_REQUESTED** [review]
Could you please get rid of binary files. They should be in gitignore.

**Comment by annaoliinyk** [comment]
Fixed

**Review by AnaKovaGH — APPROVED** [review]

**Pull request merged**